test/edu/washington/escience/myria/operator/SplitTest.java

```diff
@@ -0,0 +1,167 @@
+package edu.washington.escience.myria.operator;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.regex.PatternSyntaxException;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Longs;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleBatchBuffer;
+import edu.washington.escience.myria.util.TestEnvVars;
+
+public class SplitTest {
+
+  @Test
+  public void testGeneratedSplits() throws DbException {
+    final Object[][] expectedResults =
+        { {true, "foo", 1L, 0.1}, {true, "bar", 1L, 0.1}, {true, "baz", 1L, 0.1,},
+            {false, "", 2L, 0.2}, {false, "qux", 2L, 0.2}, {false, "", 2L, 0.2},
+            {false, "", 2L, 0.2}};
+    final Schema schema =
+        new Schema(ImmutableList.of(Type.BOOLEAN_TYPE, Type.STRING_TYPE, Type.LONG_TYPE,
+            Type.DOUBLE_TYPE), ImmutableList.of("bool", "string", "long", "double"));
```

Comment:
That's better than the parallel array approach, but still too fast and loose for my taste since the names and types can be interleaved arbitrarily :) I don't see why the API doesn't just take a set of name-type pairs--that's both clearer (to me) and safer.